screens/ChatScreen.js

```diff
@@ -59,49 +56,59 @@ class ChatScreen extends React.Component {
             selectedMessage: null,
             inputText: ''
         };
-
-        this.socket = io(baseUrl.slice(0, baseUrl.search('api')));
-        
-        this.props.createSocketConnection(this.socket);
     }
-    
+
     componentDidMount() {
-        const { 
-            navigation, 
-            createMessage, 
-            addReaction, 
-            addUserToChat, 
+        const {
+            socket,
+            fetchMessages,
+            navigation,
+            createMessage,
+            addReaction,
+            addUserToChat,
             removeUserFromChat
         } = this.props;
         const cid = navigation.getParam('chatId');
-        
-        this.props.fetchMessages(cid);
-        
-        this.socket.emit('joinChat', cid);
 
-        this.socket.on('receiveMessage', (data) => {
+        fetchMessages(cid);
+
+        socket.emit('joinChat', cid);
+
+        socket.on('receiveMessage', (data) => {
             createMessage(data);
         });
 
-        this.socket.on('receiveReaction', (data) => {
+        socket.on('receiveReaction', (data) => {
             addReaction(data);
         });
 
-        this.socket.on('receiveAddedUser', (data) => {
+        socket.on('receiveAddedUser', (data) => {
             addUserToChat(data);
         });
 
-        this.socket.on('receiveRemovedUser', (data) => {
+        socket.on('receiveRemovedUser', (data) => {
             removeUserFromChat(data);
         });
 
-        this.socket.on('removedSelf', () => {
+        socket.on('removedSelf', () => {
             this.displayToast();
         });
     }
 
     componentWillUnmount() {
-        this.socket.disconnect();
+        const { socket, currentChat } = this.props;
+        socket.emit('leaveChat', currentChat.id);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (error && prevProps.error !== error) {
```

Comment:
error is undefined here: const { error } = this.props;

Reply:
😳I'll correct that.